packages/mui-base/src/useAutocomplete/useAutocomplete.d.ts

```diff
@@ -411,7 +411,7 @@ export interface UseAutocompleteReturnValue<
    */
   getOptionProps: (
     renderedOption: UseAutocompleteRenderedOption<Value>,
-  ) => React.HTMLAttributes<HTMLLIElement>;
+  ) => React.HTMLAttributes<HTMLLIElement> & { key: any };
```

Comment:
To propagate in Base UI, no?

Reply:
would `& {key: Key}` have been more appropriate here?

The introduction of this `any` is causing me linter problems when trying to use the same pattern in my code and pass the `key` explicitly in `renderOption`.